Dockerfile

```
@@ -0,0 +1,20 @@
+FROM gcr.io/google.com/cloudsdktool/cloud-sdk
+
+RUN apt-get update && apt-get -y install npm
+
+# upgrade node to the latest version
+RUN npm install -g n && n stable
+
+WORKDIR /app
+
+# check if dependencies change, otherwise reuse layers
+COPY requirements.txt .
+RUN pip3 install -r requirements.txt
+COPY package.json .
+COPY package-lock.json .
+RUN npm install
+
+ADD . .
+RUN npm run build
```

I would at least get rid of `node_modules/` and `.git` if you're going to do it this way.

```suggestion
RUN npm run build
RUN rm -rf node_modules/ .git
```

---

`gcr.io/google.com/cloudsdktool/cloud-sdk` isn't a small image if we're worrying about size. In the Dockerfile, I was aiming toward optimizing for incremental build and run times. The image rebuilds the `node_modules` layer only when the `package.json` changes, so successive calls to `docker-compose up` to run the workflow already has dependencies installed.

We typically don't removed directories like `.git` and `site-packages` from our other dockerized applications, and it's sometimes useful to have around.